### PR TITLE
update default profile / templates to match latest artemis default configuration

### PR DESCRIPTION
**Pull request by tlbueno**
@enkeys / @michalxo this PR contains the updates related to the default profile / templates.

With these changes all default configuration that `artemis create` command currently has (2.18.0) when it generates the configuration files are now included.

Basically I tried to keep the files as close as possible with the default ones generated by artemis, so in the future we can easily spot changes in the default configuration.

Since this is my 1st PR in this repo please take a close look and let me know if you think I should change something.

PS: I didn't change the logging template/profile as I will wait to make the change in the log4j2 one.

### CODE REVIEW

**Comment by michalxo** [comment]
Personally I don't like having commented code in template (imo it does not make sense, cause you know what you are doing and what deployment you want). And you're not going to perform diff there..

Else I would say this PR is fine.

**Review by michalxo — APPROVED** [review]
Consider removing commented code. Else OK

**Comment by tlbueno** [comment]
> Personally I don't like having commented code in template (imo it does not make sense, cause you know what you are doing and what deployment you want). And you're not going to perform diff there..
> 
> Else I would say this PR is fine.

I have mixed feelings about keep the comments or not but thinking a bit more about it removing it will generate a file which is clean and simple. 

Also I just use a diff locally to ensure the default configuration options and its values are in sync with the one generated by the artemis cli create command, so, keeping the file structure in the same order but remove the comments will not make the diff check so hard since it is only comments.

So, I will remove the comments

**Comment by tlbueno** [comment]
PR updated. the comments in the files generated are removed.

File: profiles/artemis/_modules/artemis_profile/default.yaml.jinja2 [hunk 1]
```diff
@@ -6,17 +6,22 @@ artemis_profile:
...
+    - 'amq'
   java_args:
+    - -XX:AutoBoxCacheMax=20000
     - -XX:+PrintClassHistogram
     - -XX:+UseG1GC
+    - -XX:+UseStringDeduplication
     - -Xms512M
     - -Xmx2G
-    - -Xloggc:gc.log
-    - -XX:+HeapDumpOnOutOfMemoryError
```
Comment (enkeys) [inline profiles/artemis/_modules/artemis_profile/default.yaml.jinja2:17]: Removing this from the default profile is probably a good reason it's not the default. But we should probably create some other profile that allows/include it.

**Review by enkeys — CHANGES_REQUESTED** [review]

**Comment by enkeys** [comment]
> Personally I don't like having commented code in template (imo it does not make sense, cause you know what you are doing and what deployment you want). And you're not going to perform diff there..
> 
> Else I would say this PR is fine.

@michalxo @tlbueno The commented code was there because we wanted to provide a default 1:1 configuration with ActiveMQ Artemis.

File: templates/libs/headers/generator_notice.jinja2 [hunk 2]
```diff
@@ -1 +1,2 @@
-This configuration file {{ metadata.out_filename }} has been generated at {{ metadata.datetime.datetime }} by {{ metadata.tool_name }} v{{ metadata.tool_version }}
+This configuration file {{ metadata.out_filename }} has been generated
```
[the remark this answers is not in the document]
Comment (enkeys) [reply templates/libs/headers/generator_notice.jinja2:1]: never mind, now I saw that it generates a long line

File: templates/artemis/broker.xml.jinja2 [hunk 3]
```diff
@@ -42,5 +92,20 @@
         {% include 'modules/broker_xml/broker_plugins.jinja2' %}
         {% include 'modules/broker_xml/grouping_handler.jinja2' %}
         {% include 'modules/broker_xml/duplicate_cache.jinja2' %}
-    </core>
+
+      <!-- Uncomment the following if you want to use the Standard LoggingActiveMQServerPlugin pluging to log in events
```
[the remark this answers is not in the document]
Comment (tlbueno) [reply templates/artemis/broker.xml.jinja2]: I didn´t understand you comment here

File: profiles/artemis/default.yaml.jinja2 [hunk 4]
```diff
@@ -28,12 +28,10 @@ broker_xml:
 
 {% include 'artemis/_modules/broker_xml/journal/default.yaml.jinja2' %}
 
-  paging_directory: data/paging
-  bindings_directory: data/bindings
-  large_messages_directory: data/large-messages
```
[the remark this answers is not in the document]
Comment (tlbueno) [reply profiles/artemis/default.yaml.jinja2:33]: I moved it inside the journal configuration

File: templates/artemis/modules/broker_xml/queue.jinja2 [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {% from 'libs/utils.jinja2' import reformat without context %}
 {% from 'libs/utils.jinja2' import mbool without context %}
 {# DON'T TOUCH THE INDENTATION !!! #}
```
[the remark this answers is not in the document]
Comment (tlbueno) [reply templates/artemis/modules/broker_xml/queue.jinja2:3]: yes, but the formatting was not matching with the one generated by artemis cli. so, I did just a small indentation fix to match

File: templates/artemis/broker.xml.jinja2 [hunk 6]
```diff
@@ -10,23 +10,33 @@
...
+
         {% if broker_xml.global_max_size is defined %}
-            <global-max-size>{{ broker_xml.global_max_size }}</global-max-size>
+      <global-max-size>{{ broker_xml.global_max_size }}</global-max-size>
+
         {% endif %}
+        {% if broker_xml.global_max_message is defined %}
+      <global-max-messages>{{ broker_xml.global_max_message }}</global-max-messages>
 
-        {% include 'modules/broker_xml/critical_analyzer.jinja2' %}
```
[the remark this answers is not in the document]
Comment (tlbueno) [reply templates/artemis/broker.xml.jinja2:29]: just moved up in the configuration

File: profiles/artemis/_modules/artemis_profile/default.yaml.jinja2 (hunk 1 above)
Comment (tlbueno) [reply profiles/artemis/_modules/artemis_profile/default.yaml.jinja2:17]: ok. I can create a new one for this later.

File: templates/artemis/modules/broker_xml/queue.jinja2 (hunk 5 above)
[the remark this answers is not in the document]
Comment (enkeys) [reply templates/artemis/modules/broker_xml/queue.jinja2:3]: 👍

File: profiles/artemis/default.yaml.jinja2 (hunk 4 above)
[the remark this answers is not in the document]
Comment (enkeys) [reply profiles/artemis/default.yaml.jinja2:33]: 👍  Yeah, I hadn't noticed that either.

File: templates/artemis/broker.xml.jinja2 (hunk 6 above)
[the remark this answers is not in the document]
Comment (enkeys) [reply templates/artemis/broker.xml.jinja2:29]: Yeah, I hadn't noticed that either.

**Review by enkeys — APPROVED** [review]

**Comment by tlbueno** [comment]
for some reason this PR still open after merging it. So, I am going to close it.

**Pull request closed without merging**